### ISSUE TITLE
UnusedPrivateClass does not consider when class is referenced via ::class.java

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1410]
The following code produces a warning.

```kotlin
class FooBarThing {
  fun Foo.something() = getValue(Envelope::class.java)

  @Keep private data class Envelope(val data: Bar? = null)
}
```

It complains that `Envelope` is unused. It isn't though since we use it via `Envelope::class.java`

**Comment by schalkms** [comment]
Are you sure that the rule complains about the `Envelope` class and not about the data member which is unused in your sample?
Anyway, I think class references are not currently handled by this rule.

**Comment by arturbosch** [comment]
I will check this. There was a similiar false positive with class reference which should be fixed for RC12

**Comment by arturbosch** [comment]
Hey, just checked out your code but could not reproduce it. Have you tried RC12?
Checked with:
```
it("#1410") {
			val code = """
				class Foo
				class Bar
				annotation class Keep
				fun getValue(clazz: Class<*>) {
				}
				class FooBarThing {
					fun Foo.something() = getValue(Envelope::class.java)
					@Keep private data class Envelope(val data: Bar? = null)
				}

			""".trimIndent()
			val findings = UnusedPrivateClass().lint(code)
			assertThat(findings).isEmpty()
		}
```

**Comment by vanniktech** [comment]
Yeah. I'll send you the whole class privately.
